runtime/src/bank.rs

```diff
@@ -244,7 +244,7 @@ struct RentMetrics {
 pub type BankStatusCache = StatusCache<Result<()>>;
 #[cfg_attr(
     feature = "frozen-abi",
-    frozen_abi(digest = "5dfDCRGWPV7thfoZtLpTJAV8cC93vQUXgTm6BnrfeUsN")
+    frozen_abi(digest = "5UmYzdMvTDkFBKsqddQ43mSikgEA6s2bTvRZUq78YPQ2")
```

Comment:
I believe that this is actually bad, because `Result` contains `TransactionError`, which has the updated version of `TransactionError::InstructionError`, which means that snapshots that are serialized with this version of `BankSlotDelta` will be incompatible with Agave <2.3.

This _probably_ means that I have work here to do to make sure that the snapshots themselves are backward/forward compatible, but please let me know if I'm wrong about that. Maybe it's the case that we don't have an expectation that you can boot Agave 2.2 from a snapshot produced by 2.3? You tell me. cc/ @joncinque.

Reply:
Maybe @brooksprumo?

Reply:
Thanks for the guidance!

If the diff was [this](https://gist.github.com/steveluscher/25da6fc67f366d57886892316a4cbd93) (ie. adding two fields to the end of the tuple variant `InstructionError`) would that be backward compatible? That is to say if v2.3 wrote those two extra elements would v2.2 ignore them and carry on, or would it panic?

I'm pretty sure that I can get `BankSlotDeltas` to serialize like that, if that would ensure backward/forward compat, but what I'm hearing is that adding _any_ fields at all will require a SIMD no matter what.